### DISCUSSION TITLE
JavaScript Provider - OpenID Connect Protocol Mapper - Example

### DISCUSSION THREAD

**Discussion by zeschko** [keycloak/keycloak#12138] [Q&A] [answered]
Hi,

Script Mappers via GUI is removed since version 18.0.0.
I found an alternativ in the documentation:
https://github.com/keycloak/keycloak-documentation/blob/main/server_development/topics/providers.adoc

I have copied my old javascript code from the Script Mapper into a js file, build the jar file and executed the build command as described in the docs, but the mapper does not show up in the GUI.

Did the format of the javascript file changed?
There is an example for "OpenID Connect Script Authenticators", but not for "OpenID Connect Script Protocol Mappers".
Is there any example how the javascript should look like. Any imports or functions that have to be defined?

Thanks a lot!

**Comment by bonnm** [comment, marked as answer]
Hi,

The "script" feature must be enabled.

Wildfly version:
https://www.keycloak.org/docs/latest/server_installation/index.html#profiles
Quarkus version:
https://www.keycloak.org/server/features

Then the mapper is listed with its "name" you gave it in the META-INF/keycloak-scripts.json file in the mappers dropdown menu...

**Reply by zeschko** [reply]
Hi,

enabling the "scripts" feature has fixed the issue.
Thanks a lot!
